32-Refactor-Data

JWMedeiros opened the pull request:
Added more mock seed data, added dynamic seeding to prisma-client, added dynamic typing to user queries

JWMedeiros committed:
chore:  Added more mock seed data, added dynamic seeding to prisma-client, added dynamic typing to user queries

rmc3408 commented:
Missing to complete PR information.

![image](https://github.com/iPlayed-games/backend/assets/64808456/a0a2f04d-90ec-43d6-b9e6-f5e1a2ad8f91)

rmc3408 commented on src/data/mock-data.ts:
```diff
@@ -4,46 +4,120 @@ import { type Location } from '../types/location'
+  postalCode: 'L5H 1R1',
+  latitude: 12,
+  longitude: -39,
+},]
```
remove this comma line 30

JWMedeiros replied:
I ran the linter on this file, if you take the commas out it actually ends up putting the commas back in, but it spaces the ending bracket out better.

rmc3408 commented on src/data/mock-data.ts:
```diff
@@ -4,46 +4,120 @@ import { type Location } from '../types/location'
+  description: 'Come to look at the fishies, stay for the burger.',
+  logo: 'http://logo.iplayed.com',
+  mainPhoto: 'http://mainphoto.iplayed.com',
+  category: <Category[]>['Aquatic'],
```
Can you create one array with 2 items. ??

[aerial,acquatic]

JWMedeiros replied:
I added this and it does in fact work. I added second categories for venues 2 and 3.

rmc3408 commented on src/data/mock-data.ts:
```diff
@@ -4,46 +4,120 @@ import { type Location } from '../types/location'
+  mainPhoto: 'http://mainphoto.iplayed.com',
+  category: <Category[]>['Aerial'],
+  isAcessibility: false,
+  workingHoursOpen: 1692349200000, 
```
can you change to a new time for future testing ? this is epoch time 

workingHoursOpen and workingHoursClose

JWMedeiros replied:
I changed the working hours for the second and third venus, and also changed activity times for activities 2 and 3 for testing.

rmc3408 commented on src/config/prisma/prisma-client.ts:
```diff
@@ -15,39 +15,40 @@ export async function startORM() {
-                    },
-                  },
-                ],
+      for (let i=0; i<userData.length; i++){
```
space between i<userData.

JWMedeiros replied:
Linter cleaned that up as well.

rmc3408 reviewed (commented):
LGTM. Just few changes some data info for us in the future and tiny formatting issues.

rmc3408 commented on src/config/prisma/prisma-client.ts:
```diff
@@ -15,39 +15,40 @@ export async function startORM() {
-      })
-
-      // await prisma.$disconnect()
-      // shutDown('database successful seeded')
```
Add this code back and running please.

JWMedeiros replied:
Its added back in. Resolving the conversation for this.

rmc3408 reviewed (commented):
Please add back the code for SHUTDOWN and running (un-comment out)

JWMedeiros committed:
Update prisma-client.ts

Added back in shutdown code after DB is seeded.

JWMedeiros committed:
Update prisma-client.ts

Added spaces in test use code.

JWMedeiros committed:
Update mock-data.ts

Ran the linter, changed the times for testing, added second category for venue 2 and 3.

rmc3408 reviewed (approved):
LGTM , ready to merge
